lib/Net/SAML2/IdP.pm

```diff
@@ -290,13 +314,28 @@ sub art_url {
 
 =head2 cert( $use )
 
-Returns the IdP's certificate for the given use (e.g. C<signing>).
+Returns the IdP's certificates for the given use (e.g. C<signing>).
+
+IdP's are generated from the metadata it is possible for multiple certificates
+to be contained in the metadata and therefore possible for them to be there to
+be multiple verified certs in $self->certs.  At this point any certs in the IdP
+have been verified and are valid for the specified use.  All certs are of type
+$use are returned.
 
 =cut
 
 sub cert {
     my($self, $use) = @_;
-    return $self->certs->{$use};
```

Comment:
If we would use the hashref here it would automatically return the array ref. Removing all the logic you added below.